lib/improver/tests/ensemble_calibration/ensemble_calibration/test_EnsembleCalibration.py

```diff
@@ -314,82 +626,104 @@ def test_wind_speed_realizations_data_check(self):
+            [[1.3333356, 2.0000021, 2.6666684],
+             [3.3333352, 4.0000014, 4.666668],
+             [5.3333344, 6.000001, 6.6666675]], dtype=np.float32)
+        expected_wind_speed_variance_data = np.array(
+            [[0., 0., 0.],
+             [0., 0., 0.],
+             [0., 0., 0.]], dtype=np.float32)
         distribution = "truncated gaussian"
         desired_units = "m s^-1"
         predictor_of_mean_flag = "realizations"
         plugin = Plugin(
-            calibration_method, distribution, desired_units,
-            predictor_of_mean_flag=predictor_of_mean_flag)
+            self.calibration_method, distribution, desired_units,
+            predictor_of_mean_flag=predictor_of_mean_flag,
+            max_iterations=10000)
         calibrated_predictor, calibrated_variance = plugin.process(
             self.current_wind_speed_forecast_cube,
             self.historic_wind_speed_forecast_cube,
             self.wind_speed_truth_cube)
-        self.assertArrayAlmostEqual(calibrated_predictor.data, predictor_data,
-                                    decimal=4)
-        self.assertArrayAlmostEqual(calibrated_variance.data, variance_data)
+        self.assertArrayAlmostEqual(
+            calibrated_predictor.data, expected_wind_speed_predictor_data,
+            decimal=3)
+        self.assertArrayAlmostEqual(
+            calibrated_variance.data, expected_wind_speed_variance_data,
+            decimal=3)
+
+
+@unittest.skipIf(
+    STATSMODELS_FOUND is True, "statsmodels module is available.")
+class Test_process_without_statsmodels(SetupCubes):
+
+    """Additional tests for the process method when the statsmodels module
+    is not available. A simple initial guess will be used for the
+    coefficients that will be solved as part of the calibration."""
 
     @ManageWarnings(
         ignored_messages=IGNORED_MESSAGES, warning_types=WARNING_TYPES)
-    def test_alternative_calibration_name(self):
+    def test_temperature_realizations_data_check(self):
         """
-        Test that the plugin returns the calibrated predictor and the
-        calibrated variance if an alternative name for the calibration
-        is provided. The ensemble mean is the predictor.
+        Test that the plugin returns an iris.cube.CubeList
+        of temperature cubes with the expected data, where the plugin
+        returns a cubelist of, firstly, the predictor and, secondly the
+        variance.
+        The ensemble realizations is the predictor.
         """
-        calibration_method = "nonhomogeneous gaussian regression"
         distribution = "gaussian"
         desired_units = "degreesC"
-        plugin = Plugin(calibration_method, distribution, desired_units)
+        predictor_of_mean_flag = "realizations"
+        plugin = Plugin(
+            self.calibration_method, distribution, desired_units,
+            predictor_of_mean_flag=predictor_of_mean_flag,
+            max_iterations=10000)
```

Comment:
Do you need this to be this high when statsmodels is available?

Reply:
I've reduced the number of max iterations when using statsmodels.